sbt-plugin/src/main/scala-sbt-1.0/org/scalajs/sbtplugin/SBTCompat.scala

```diff
@@ -0,0 +1,68 @@
+package org.scalajs.sbtplugin
+
+import sbt._
+
+import org.scalajs.core.tools.io.FileVirtualFile
+
+private[sbtplugin] object SBTCompat {
+  type IncOptions = xsbti.compile.IncOptions
+
+  type CompileAnalysis = xsbti.compile.CompileAnalysis
+
+  val formatImplicits: sjsonnew.BasicJsonProtocol.type =
+    sjsonnew.BasicJsonProtocol
+
+  def moduleIDWithConfigurations(moduleID: ModuleID,
+      configurations: Option[String]): ModuleID = {
+    moduleID.withConfigurations(configurations)
+  }
+
+  def crossVersionAddScalaJSPart(cross: CrossVersion,
+      part: String): CrossVersion = {
+    cross match {
+      case CrossVersion.Disabled =>
+        CrossVersion.constant(part)
+      case cross: sbt.librarymanagement.Constant =>
+        cross.withValue(part + "_" + cross.value)
+      case cross: CrossVersion.Binary =>
+        cross.withPrefix(part + "_" + cross.prefix)
+      case cross: CrossVersion.Full =>
+        cross.withPrefix(part + "_" + cross.prefix)
```

Comment:
They syntactically do the same thing. But in the `case Binary`, `cross` is a `Binary` so `cross.withPrefix` constructs a `Binary`. Whereas in `case Full`, `cross.withPrefix` constructs a `Full`.